lnwallet/channel.go

```diff
@@ -3892,16 +3892,30 @@ func (lc *LightningChannel) ProcessChanSyncMsg(
 			return nil, nil, nil, err
 		}
 
+		var commitUpdates []lnwire.Message
+
 		// Next, we'll need to send over any updates we sent as part of
 		// this new proposed commitment state.
 		for _, logUpdate := range commitDiff.LogUpdates {
-			updates = append(updates, logUpdate.UpdateMsg)
+			commitUpdates = append(commitUpdates, logUpdate.UpdateMsg)
 		}
 
 		// With the batch of updates accumulated, we'll now re-send the
 		// original CommitSig message required to re-sync their remote
 		// commitment chain with our local version of their chain.
-		updates = append(updates, commitDiff.CommitSig)
+		commitUpdates = append(commitUpdates, commitDiff.CommitSig)
+
+		// NOTE: If a revocation is not owed, then updates is empty.
+		if lc.channelState.LastWasRevoke {
+			// If lastWasRevoke is set to true, a revocation was last and we
+			// need to reorder the updates so that the revocation stored in
+			// updates comes after the LogUpdates+CommitSig.
+			updates = append(commitUpdates, updates...)
```

Comment:
I find this a bit hard to follow, possibly because we are assuming something about the updates that have been added in an earlier switch statement.

Could an alternative be to split this up
```go
var owedRevocation, owedCommitUpdates []lnwire.Message

switch {
case ...:
    owedRevocation = append(owedRevocation, revokeAndAck)
}

switch {
case ...:
    owedCommitUpdates = append(owedCommitUpdates, commitSig)
}

var updates []lnwire.Message
if lastWasRevoke {
    updates = append(owedCommitUpdates, owedRevocation)
else {
    updates = append(owedRevocation, owedCommitUpdates) 
}

```

Reply:
I chose this way because this does not account for `OweCommitment`.  Would it then be added to `owedRevocation`? - Then the name doesn't describe what it does.  The switch `case msg.NextLocalCommitHeight == remoteTipHeight` cannot be called if `OweCommitment` is called so that doesn't change our current logic if a revocation exists or not.

Reply:
> I chose this way because this does not account for OweCommitment

So won't this end up _re ordering_ the `OweCommitment` sig if we need to generate+send it?

Reply:
> The switch case msg.NextLocalCommitHeight == remoteTipHeight cannot be called if OweCommitment is called so that doesn't change our current logic if a revocation exists or not.

Ah ok, make sense.